packages/ra-core/src/form/useIsFormInvalid.ts

```diff
@@ -20,9 +20,15 @@ export const useIsFormInvalid = () => {
         if (submitCount > submitCountRef.current) {
             submitCountRef.current = submitCount;
 
-            setIsInvalid(!isValid);
+            // For some reason, the validation state might not be sync yet on first submit
```

Comment:
Isn't this related to the fact that `isValid` is affected by mode at useForm. This state is only applicable with onChange, onTouched, and onBlur mode.

https://react-hook-form.com/api/useform

Reply:
Good catch, the fix is ok though as it will work in any mode (which users can change)